analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCognitiveComplexityMetric.cs

```diff
@@ -31,19 +32,27 @@ namespace SonarAnalyzer.Metrics.CSharp
 {
     public static class CSharpCognitiveComplexityMetric
     {
-        public static CognitiveComplexity GetComplexity(SyntaxNode node)
+        public static CognitiveComplexity GetComplexity(SyntaxNode node) =>
+            GetComplexity(node, false);
+
+        public static CognitiveComplexity GetComplexity(SyntaxNode node, bool onlyGlobalStatements)
         {
-            var walker = new CognitiveWalker();
+            var walker = new CognitiveWalker(onlyGlobalStatements);
             walker.SafeVisit(node);
 
             return new CognitiveComplexity(walker.State.Complexity, walker.State.IncrementLocations.ToImmutableArray());
         }
 
         private class CognitiveWalker : CSharpSyntaxWalker
         {
+            private readonly bool onlyGlobalStatements;
+
             public CognitiveComplexityWalkerState<MethodDeclarationSyntax> State { get; }
                 = new CognitiveComplexityWalkerState<MethodDeclarationSyntax>();
 
+            public CognitiveWalker(bool onlyGlobalStatements) =>
+                this.onlyGlobalStatements = onlyGlobalStatements;
```

Comment:
Coding Style: This could go in one line to increase readability.

Reply:
AFAIK, this is considered to be "complex enough" to be on the new line. 

So we are having one liners only in case of => false or  => 42, which are indeed easy ones